test/test_ldap_connection.rb

```diff
@@ -291,7 +291,7 @@ def test_queued_read_setup_encryption_with_start_tls
       and_return(result2)
     mock.should_receive(:write)
     conn = Net::LDAP::Connection.new(:socket => mock)
-    flexmock(Net::LDAP::Connection).should_receive(:wrap_with_ssl).with(mock, {}).
+    flexmock(Net::LDAP::Connection).should_receive(:wrap_with_ssl).with(mock, {}, nil).
```

Comment:
@tmm1 Doesn't this mocking cover your changes on `wrap_with_ssl`, do ti? If you tested your changes work as you expected, could you share how to do it? 

I know it's really hard to reproduce the situation timeout or something network issues happens.